File: Herança - Subclasses de Personagem/Mago.py
class Mago(Personagem):
    def __init__(self, nome, vida, forca, nivel):
        super().__init__(nome, vida, forca, nivel)
        self.setMana(50)

    def getMana(self):
        return self.mana

    def setMana(self, m):
        if m >= 0:
            self.mana = m
        else:
            print("Valor de mana inválido.")

    def ficha(self):
        print("\n====FICHA==== \nNome: " + self.getNome() +
              "\nVida: " + str(self.getVida()) +
              "\nForça: " + str(self.getForca()) +
              "\nNivel: " + str(self.getNivel()) +
              "\nMana: " + str(self.getMana()) +
              "\nInventario:")

        if len(self.getInventario()) == 0:
            print("Inventario Vazio.")
        else:
            for i in range(len(self.getInventario())):
                print(self.getInventario()[0].descricao())

        print("===========")

File: Herança - Subclasses de Personagem/Mago.java
class Mago extends Personagem{
    private int mana;

    public Mago(String nome, int vida, int forca, int nivel){
        super(nome, vida, forca, nivel);
        setMana(50);
    }

    public int getMana(){
        return this.mana;
    }
    public void setMana(int m){
        if (m>=0){
            this.mana = m;
        }
        else{
            System.out.println("Valor de mana inválido.");
        }
    }
    @Override
    public void ficha(){
        System.out.println("\n====FICHA==== \nNome: " + getNome() + "\nVida: " + getVida() + "\nForça: " + getForca() + "\nNivel: " + getNivel() + "\nMana: " + getMana() + "\nInventario:");
        if (getInventario().size() == 0){
            System.out.println("Inventario Vazio.");
        }
        else{
            for (int i=0; i<getInventario().size(); i++){
                System.out.println(getInventario().get(0).descricao());
            }
        }
        System.out.println("===========");
    }
}